### DISCUSSION TITLE
Alien-Build 7430->7412: Entpacken der Hersteller-Firmware schlägt fehl

### DISCUSSION THREAD

**Discussion by mparthey** [Freetz-NG/freetz-ng#1020] [Allgemein] [answered]
Wenn ich versuche, die Freetz-NG-Firmware der 7430 als Alien für die 7412 zu bauen, dann scheitert das Entpacken der Hersteller-Firmware für die 7412 (Version 06.88):
```
STEP 1B: UNPACK 2ND
source firmware file: dl/fw/FRITZ.Box_7412.137.06.88.image
checking signature: valid
unpacking firmware image
unpacking outer-filesystem image
ERROR: modunext2: Error in build/.tk/original/firmware/var/tmp/filesystem.image
ERROR: unpacking failed
FINISHED

```
Dieser Fehler ähnelt einem [anderen](https://github.com/Freetz-NG/freetz-ng/discussions/849), der bereits behoben wurde, ist aber nicht der gleiche.

Build-Umgebung:
Fedora 38 mit Freetz prerequisites
Freetz revision: freetz-ng 24239-22ddd68e8 master 2024-07-30

make menuconfig:
```
FREETZ_USER_LEVEL_EXPERT=y
FREETZ_SERIES_4=y
FREETZ_TYPE_7430=y
FREETZ_TYPE_ALIEN_HARDWARE=y
FREETZ_TYPE_7412_7430=y

```
[dot_config.gz](https://github.com/user-attachments/files/16455048/dot_config.gz)

**Comment by fda77** [comment]
Beim verlinkten Problem schlug die Signaturprüfung fehl, weil via Juis die Firmware eines anderen Gerätes (alien) verwendet wurde. Nach bfa279fd21cd988ce1c4408b1a977509bdc13108 geht das.

Ich hab mit deiner config auf Fedora 40 compiliert und es lief durch:

![image](https://github.com/user-attachments/assets/2f2c7ef8-286a-4d9a-973e-870dcea7b1cb)

Daher weiss ich nicht so recht warum es bei dir nicht funktioniert. Da deine Signatur okay ist sollte der Dateiinhalt auch unbeschädigt sein.

**Comment by MasterRoCcO** [comment, marked as answer]
Man kann das Problem umgehen, in dem man erst ein Image für die 7412 erstellt und dann erst das Alien Image erstellt. Der Fehler ist seit 2023 schon bekannt.

Sorry 2022.

**Reply by mparthey** [reply]
> Man kann das Problem umgehen, in dem man erst ein Image für die 7412 erstellt und dann erst das Alien Image erstellt.

Vielen Dank für diesen Tip, damit konnte ich das Alien Image jetzt erfolgreich erstellen.

**Comment by fda77** [comment]
was hab ich dann "falsch" gemacht damit es ohne fehler geht?

**Comment by fda77** [comment]
loift:
https://github.com/Freetz-NG/freetz-ng/actions/runs/10200617594
https://github.com/Freetz-NG/freetz-ng/actions/runs/10200617594/job/28220538154

**Comment by mparthey** [comment]
Ich habe jetzt ein frisches Fedora 40 installiert und versuche damit `Freetz-NG 24246-ddd3c9dc0 master 2024-07-31` zu bauen.
Der Schritt `---> tools/prelink-host` bricht ab mit dieser Meldung:
```
configure: error: libelf does not properly convert Elf64_Sxword quantities.
If you are using libelf-0.7.0, please use patches/libelf-0.7.0.patch.
```
Logfiles kann ich bei Bedarf nachliefern.

**Comment by PaulchenPlump** [comment]
Habe auch das libelf-Problem nach der Bereinigung meiner Installation mittels `make dirclean`.

**Comment by fda77** [comment]
ich habe keine ahnung
https://github.com/Freetz-NG/freetz-ng/actions/workflows/dl-hosttools.yml
https://github.com/Freetz-NG/freetz-ng/actions/runs/10148750730/job/28062170356
bei anderen gehts

**Comment by mparthey** [comment]
Hier noch die [Ausgabe von make](https://github.com/user-attachments/files/16469892/make-master-ddd3c9dc0-typescript.gz), aufgezeichnet mit `script`, sowie das [config.log von prelink](https://github.com/user-attachments/files/16469914/make-master_host-tools_prelink-20130503_config.log.gz).

In der Ausgabe finden sich ein paar Warnungen. Ich weiß nicht, ob die von Bedeutung sind:
```
autoreconf: warning: autoconf input should be named 'configure.ac', not 'configure.in'
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
m4/libelf.m4:8: warning: underquoted definition of AC_LIBELF_SXWORD
m4/libelf.m4:8:   run info Automake 'Extending aclocal'
m4/libelf.m4:8:   or see https://www.gnu.org/software/automake/manual/automake.html#Extending-aclocal

```

**Reply by mparthey** [reply]
Ich habe jetzt wieder auf Debian buster gebaut, vorsichtshalber mit FREETZ_HOSTTOOLS_DOWNLOAD=n, das hat geklappt.

Das Fedora 40 läuft auf einem anderen Rechner unter qemu-kvm. Eventuell war der Auslöser für das Problem mit prelink die ältere CPU oder die Virtualisierung. Das ist aber reine Spekulation, ich habe keinen Vergleichstest gemacht.

**Reply by fda77** [reply]
FREETZ_HOSTTOOLS_DOWNLOAD=n  muss sein damit prelink überhaupt gebaut wird

Was die warnings angeht: das ist normal die Sache von dem Programm das diese ausgibt
